java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitutionSolver.java

```diff
@@ -166,8 +162,8 @@ private JavaType substituteInArrayType(JavaType.ArrayJavaType arrayType, TypeSub
       rootElementType = ((JavaType.ArrayJavaType) rootElementType).elementType;
       nbDimensions++;
     }
-    JavaType substitutedType = substitution.substitutedType(rootElementType);
-    if (substitutedType != null) {
+    JavaType substitutedType = applySubstitution(rootElementType, substitution);
+    if (substitutedType != rootElementType) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)